private-templates-service/adaptivecards-templating-client/typescript/api.ts

```diff
@@ -14,1487 +14,1343 @@ import localVarRequest = require('request');
-          return data[discriminatorProperty]; // use the type given in the discriminator
+    public static findCorrectType(data: any, expectedType: string) {
+        if (data == undefined) {
+            return expectedType;
+        } else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
+            return expectedType;
+        } else if (expectedType === "Date") {
+            return expectedType;
         } else {
-          return expectedType; // discriminator was not present (or an empty string)
+            if (enumsMap[expectedType]) {
+                return expectedType;
+            }
+
+            if (!typeMap[expectedType]) {
+                return expectedType; // w/e we don't know the type
+            }
+
+            // Check the discriminator
+            let discriminatorProperty = typeMap[expectedType].discriminator;
+            if (discriminatorProperty == null) {
+                return expectedType; // the type does not have a discriminator. use it.
+            } else {
+                if (data[discriminatorProperty]) {
+                    return data[discriminatorProperty]; // use the type given in the discriminator
+                } else {
+                    return expectedType; // discriminator was not present (or an empty string)
+                }
+            }
         }
-      }
     }
-  }
-
-  public static serialize(data: any, type: string) {
-    if (data == undefined) {
-      return data;
-    } else if (primitives.indexOf(type.toLowerCase()) !== -1) {
-      return data;
-    } else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
-      let subType: string = type.replace("Array<", ""); // Array<Type> => Type>
-      subType = subType.substring(0, subType.length - 1); // Type> => Type
-      let transformedData: any[] = [];
-      for (let index in data) {
-        let date = data[index];
-        transformedData.push(ObjectSerializer.serialize(date, subType));
-      }
-      return transformedData;
-    } else if (type === "Date") {
-      return data.toString();
-    } else {
-      if (enumsMap[type]) {
-        return data;
-      }
-      if (!typeMap[type]) { // in case we dont know the type
-        return data;
-      }
-
-      // get the map for the correct type.
-      let attributeTypes = typeMap[type].getAttributeTypeMap();
-      let instance: { [index: string]: any } = {};
-      for (let index in attributeTypes) {
-        let attributeType = attributeTypes[index];
-        instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
-      }
-      return instance;
+
+    public static serialize(data: any, type: string) {
+        if (data == undefined) {
+            return data;
+        } else if (primitives.indexOf(type.toLowerCase()) !== -1) {
+            return data;
+        } else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
```

Comment:
are those just a formatting chaneges?

Reply:
Yep, the files under `adaptivecards-templating-client/typescript` were autogenerated from the `swagger.yaml` file, I've updated the description to include this as well!